src/control/destination/background/destination_set.rs

```diff
@@ -35,6 +35,13 @@ pub(super) struct DestinationSet<T: HttpService<ResponseBody = RecvBody>> {
     pub query: Option<DestinationServiceQuery<T>>,
     pub dns_query: Option<IpAddrListFuture>,
     pub responders: Vec<Responder>,
+
+    /// This flag is set if `query` is `None` because we wanted to make a
+    /// query for this destination, but were at the query capacity limit.
+    ///
+    /// Otherwise, if `query` is `None` and this is _not_ set, then the
+    /// authority does not require a Destination query.
```

Comment:
This has the added benefit of simplifying return types in NewQuery

Reply:
Yeah, that might be better. I think there's at least one case where we `take()` the `query` field, so that would need to be rewritten, but I definitely like your suggestion more than the `Option` + `bool` situation...

Reply:
could also do something along the lines of

```rust
enum LimitedCapacity<T> { Available(T), Unavailable }
```

and then return `LimitedCapacity<Option<...>>`.  not sure if that's better or worse

but, i think encoding this in the type system one way or the other will be easier to think about

Reply:
You were right, the enum did in fact end up being the right way to represent this. Changed in e7f79dc066a8d87404966ac044a23196cf4273fe.